src/App.tsx

```diff
@@ -1,26 +1,389 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
+import * as todoService from './api/todos';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { TempTodo } from './components/TempTodo';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  // for showing loading overlay while adding todo
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  // for input disabling
+  const [isTodoLoading, setIsTodoLoading] = useState(false);
+  // for showing loading overlay while deleting or updating todo
+  const [loadingTodoId, setLoadingTodoId] = useState(0);
+  // for redacting todos
+  const [selectedTodo, setSelectedTodo] = useState<Todo | undefined>(undefined);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  //#region effects
+  // hide error message in 3 secs
+  useEffect(() => {
+    if (errorMessage.length !== 0) {
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, [errorMessage]);
+
+  // load todos 1 time on page loading
+  useEffect(() => {
+    setIsTodoLoading(true);
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage(`Unable to load todos`))
+      .finally(() => setIsTodoLoading(false));
+  }, []);
+  //#endregion
+
+  //#region filter Todos
+  const [selectedFilter, setSelectedFilter] = useState<Filter>('All');
+
+  function getVisibleTodos() {
+    let visibleTodos: Todo[];
+
+    switch (selectedFilter) {
+      case 'All':
+        visibleTodos = todos;
+        break;
+      case 'Active':
+        visibleTodos = todos.filter(todo => !todo.completed);
+        break;
+      case 'Completed':
```

Comment:
Create a enum for 'All', 'Active', 'Completed' and use it everywhere

Reply:
I already created type 'Filter' and put it in types directory, check selectedFilter state, it has this exact type